tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -37,7 +38,27 @@ export enum BinaryOpType {
   COMPLEX_MULTIPLY_IMAG
 }
 
+const CHECK_NAN_SNIPPET = `
+  if (isnan(a)) { return a; }
+  if (isnan(b)) { return b; }
+  `;
+const CHECK_NAN_SNIPPET_VEC4 = `
+  if (isNaN.r) {
+    resultTemp.r = uniforms.NAN;
+  }
+  if (isNaN.g) {
+    resultTemp.g = uniforms.NAN;
+  }
+  if (isNaN.b) {
+    resultTemp.b = uniforms.NAN;
+  }
+  if (isNaN.a) {
+    resultTemp.a = uniforms.NAN;
+  }
+  `;
+
 const ADD = 'return a + b;';
+const ATAN2 = 'return atan2(a, b);';
```

Comment:
@pyu10055 @lina128   Do you know when `NaN` check will be needed? I notice that webgl's [`ATAN2`](https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgl/src/kernels/Atan2.ts#L23) has this checking. However, not all kernels have this checking. And  it seems that no test case covers it for `Atan2`. So I am wondering is there a rule to describe which ops need such kind of checking? And personally, I think `isnan` check is a burden for gpu. We should try to avoid it as much as possible.

Reply:
Discussed it internally. One idea in our side is only do NaN check behind a flag. And @axinging will do the refactor in a separate PR. @pyu10055 @lina128 Please let us know if you have more opinions. Will merge this PR firstly. Thanks.